demo/frontend/src/App.tsx

```diff
@@ -304,15 +231,128 @@ const App = () => {
+          >
+            Send Embrace Error Log
+          </button>
+        </div>
+
+        {/* Properties: */}
+        <div className={styles.actions}>
+          <button
+            onClick={() =>
+              handleAddPermanentSessionProperty(
+                'permanent-key',
+                'permanent-value'
+              )
+            }
+            disabled={sessionProvider.getSessionSpan() === null}
+          >
+            Add Permanent Session Property
+          </button>
+
+          <button
+            onClick={() => handleRemoveSessionProperty('permanent-key')}
+            disabled={sessionProvider.getSessionSpan() === null}
+          >
+            Remove Permanent Session Property
+          </button>
+        </div>
+        <div className={styles.actions}>
+          <button
+            onClick={() =>
+              handleAddSessionProperty('session-key', 'session-value')
+            }
+            disabled={sessionProvider.getSessionSpan() === null}
+          >
+            Add Session Property
+          </button>
+          <button
+            onClick={() => handleRemoveSessionProperty('session-key')}
+            disabled={sessionProvider.getSessionSpan() === null}
+          >
+            Remove Session Property
+          </button>
+        </div>
+
+        {/* Exceptions: */}
+        <div className={styles.actions}>
+          <button
+            onClick={handleRecordException}
+            disabled={sessionProvider.getSessionSpan() === null}
+          >
+            Record Exception
+          </button>
+          <button
+            onClick={handleThrowError}
+            disabled={sessionProvider.getSessionSpan() === null}
+          >
+            Throw Error
+          </button>
+          <button
+            onClick={handleRejectPromise}
+            disabled={sessionProvider.getSessionSpan() === null}
+          >
+            Reject Promise
+          </button>
+        </div>
+
+        {/* Network: */}
+        <div className={styles.actions}>
+          <button onClick={handleSendFetchNetworkRequest}>
+            Send a Fetch Network Request
+          </button>
+          <button onClick={handleSendFetchNetworkRequest404}>
+            Send a Fetch Network Request (404)
+          </button>
+          <button onClick={handleSendXMLNetworkRequest}>
+            Send a XML Network Request
+          </button>
+        </div>
+
+        {/* React: */}
+        <div className={styles.actions}>
+          <button onClick={() => setNavigationType('declarativeV4V5')}>
+            Enter react-router v4/v5 navigation demo
+          </button>
+          <button onClick={() => setNavigationType('declarativeV6+')}>
+            Enter react-router v6+ declarative navigation demo
+          </button>
+          <button onClick={() => setNavigationType('data')}>
+            Enter react-router v6+ data navigation demo
+          </button>
+        </div>
+
+        {/* Navigation: */}
```

Comment:
Added this section to test how the sdk behaves when the app navigates. Tested mostly the opening on a new tab with cmd+tab very consecutive and couldn't get any session_number that overlaps so far